justfile

```diff
@@ -16,60 +18,81 @@ linter-setup:
   source {{hermitActivationScript}} || true
   pre-commit install -f
 
-# Vanilla
-
 create-vanilla-deployment-file:
-   kustomize build {{kustomizeVanillaOverlay}} --output deploy/k8s/cloudbeat-ds.yaml
-
-build-deploy-cloudbeat: build-cloudbeat load-cloudbeat-image deploy-cloudbeat
-
-build-load-both: build-deploy-cloudbeat load-pytest-kind
-
-build-deploy-cloudbeat-debug: build-cloudbeat-debug load-cloudbeat-image deploy-cloudbeat
-
-build-replace-cloudbeat: build-binary
+  @echo "Make sure to run 'eval \$(elastic-package stack shellinit --shell \$(basename $SHELL))'"
+  cp {{env_var('ELASTIC_PACKAGE_CA_CERT')}} {{kustomizeVanillaOverlay}}
+  kustomize build {{kustomizeVanillaOverlay}} --output deploy/k8s/cloudbeat-ds.yaml
+
+build-deploy-cloudbeat $GOOS=LOCAL_GOOS $GOARCH=LOCAL_GOARCH:
+  just build-cloudbeat-docker-image $GOOS $GOARCH
+  just load-cloudbeat-image
+  just deploy-cloudbeat
+
+build-deploy-cloudbeat-debug $GOOS=LOCAL_GOOS $GOARCH=LOCAL_GOARCH:
+  just build-cloudbeat-debug $GOOS $GOARCH
+  just load-cloudbeat-image
+  just deploy-cloudbeat
+
+# Builds cloudbeat binary and replace it in the agents (in the current context - i.e. `kubectl config current-context`)
+# Set GOOS and GOARCH to the desired values (linux|darwin, amd64|arm64)
+build-replace-cloudbeat $GOOS=LOCAL_GOOS $GOARCH=LOCAL_GOARCH: build-opa-bundle
+  just build-binary $GOOS $GOARCH
+  # replace the cloudbeat binary in the agents
   ./scripts/remote_replace_cloudbeat.sh
-
-build-replace-bundle: build-opa-bundle
+  # Replace bundle in the agents
   ./scripts/remote_replace_bundle.sh
 
 load-cloudbeat-image kind='kind-multi':
   kind load docker-image cloudbeat:latest --name {{kind}}
 
 build-opa-bundle:
+  go mod vendor
   mage BuildOpaBundle
 
-build-binary:
-  GOOS=linux go mod vendor
-  GOOS=linux go build -v
+# Builds cloudbeat binary
+# Set GOOS and GOARCH to the desired values (linux|darwin, amd64|arm64) by default, it will use local platform
+build-binary $GOOS=LOCAL_GOOS $GOARCH=LOCAL_GOARCH:
+  @echo "Building cloudbeat binary for $GOOS/$GOARCH"
+  go mod vendor
+  go build -v
+
+# For backwards compatibility
+alias build-cloudbeat := build-cloudbeat-docker-image
 
-build-cloudbeat: build-opa-bundle build-binary
-  docker build -t cloudbeat .
+# Builds cloudbeat docker image with the OPA bundle included
+# Set GOOS and GOARCH to the desired values (linux|darwin, amd64|arm64)
+build-cloudbeat-docker-image $GOOS=LOCAL_GOOS $GOARCH=LOCAL_GOARCH: build-opa-bundle
+  just build-binary $GOOS $GOARCH
+  @echo "Building cloudbeat docker image for $GOOS/$GOARCH"
+  docker build -t cloudbeat . --platform=$GOOS/$GOARCH
```

Comment:
sounds good, setting target OS as `linux` makes more sense. Regarding renaming suggestion, I would prefer your option also, but I chose `GOOS` and `GOARCH` because these are the prams that `go build` uses, so having the same name will mean we don't need to send them explicitly. This is also open to discussion though, if you think we should send them explicitly I can rename them to `OS` and `ARCH`

By the way, you can use multi-line code blocks with `\`\`\``